jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt

```diff
@@ -116,6 +117,12 @@ internal sealed class CodeScanSessionConfig(
         false -> "${getPayloadLimitInBytes() / TOTAL_BYTES_IN_KB}KB"
     }
 
+    open fun getTotalProjectSizeInBytes(): Long = VfsUtil.collectChildrenRecursively(projectRoot).filter {
```

Comment:
how does ` VfsUtil.collectChildrenRecursively` handle timeout here ?

Reply:
The event is sent after the codescan results are displayed. So it should be fine even if it takes a little longer. Nonetheless, I have tried on some fairly large projects (~2.2 GB) and it works within milliseconds.  

Reply:
The timeout config is not just for codescan job, but also prevent failure modes e.g. recursion conduct outside target folder by some bug, it might end up with scanning the customer's all system files.

Reply:
What kind of bug? I cannot think of a scenario where it might scan customer's system files outside of project.